common/src/main/java/com/wynntils/features/user/UpdateReminderFeature.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.webapi.request.Request;
+import com.wynntils.core.webapi.request.RequestBuilder;
+import com.wynntils.core.webapi.request.RequestHandler;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.WorldStateManager;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class UpdateReminderFeature extends UserFeature {
+    private static final String CI_LINK =
+            "https://ci.wynntils.com/job/Artemis/lastSuccessfulBuild/api/json?tree=number";
+    private static final String DOWNLOAD_LINK = "https://ci.wynntils.com/job/Artemis";
+
+    private boolean firstJoin = true;
+
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent event) {
+        if (event.getNewState() == WorldStateManager.State.NOT_CONNECTED) {
+            firstJoin = false;
+            return;
+        }
+
+        if (event.getNewState() != WorldStateManager.State.WORLD || !firstJoin) return;
+
+        firstJoin = false;
+
+        Request versionFetch = new RequestBuilder(CI_LINK, "latest_build")
```

Comment:
If you want to prepare more for an automatic updating service, perhaps try to extract the functionality that checks if there is a newer version into a "UpdateUtils" (or whatever), instead of coding that logic into the Feature.

Reply:
We would want auto update eventually. I am trying not to add it until we add the GH CI changes.

Reply:
I feel that extraction should be done right now, so it isnt forgotten to be done later.

Reply:
But if we move to github, all of this code has to be removed anyways.

Reply:
Ok, fine, it's a good start.